service/raft/raft_group0.cc

```diff
@@ -1455,9 +1477,8 @@ future<> raft_group0::do_upgrade_to_group0(group0_upgrade_state start_state) {
         // a very high chance that the following steps succeed as well (we would need to be very unlucky otherwise).
         upgrade_log.info("Performing a dry run of remote `get_group0_upgrade_state` calls...");
         co_await check_remote_group0_upgrade_state_dry_run(
-                [&group0_server] {
-                    auto current_config = get_raft_members_inet_addrs(group0_server.get_configuration().current);
-                    return make_ready_future<std::vector<gms::inet_address>>(std::move(current_config));
+                [members0] {
+                    return make_ready_future<std::vector<gms::inet_address>>(members0.get_inet_addrs());
```

Comment:
Is it guaranteed that get_inet_addrs() is non-empty here?

Reply:
Non-emptiness is checked at the call site (in `check_remote_group0_upgrade_state_dry_run`), if it's empty it retries

Reply:
Ok, fine then.